lib/rules/base.js

```diff
@@ -65,6 +68,26 @@ module.exports = class Base {
     this.source = options.rawSource.match(reLines);
   }
 
+  get filePath() {
+    return this._filePath;
+  }
```

Comment:
🤔 why not change the class field to be `this.filePath = options.filePath` instead of adding a getter?

Reply:
lol, I'm looking forward to support deprecation warns once accessed xD

Reply:
and we already have `_filePath` (keeping backward compability)

Reply:
Ya, but the deprecation would be for `_filePath` in this case since conceptually `filePath` is the main go forward / correct value to use (as opposed to `moduleId`/`moduleName`).

Let's do this:

```js
// preserve `this._filePath` compatibility for now, even though it is
// private API (future versions may deprecate)
this._filePath = this.filePath = options.filePath;
```

Then use a private symbol + getters for `_moduleId` + `_moduleName`, (you are right that this makes future deprecation fairly painless).

```js
const MODULE_ID = Symbol('_moduleId');
const MODULE_NAME = Symbol('_moduleName');

class Rule {
  constructor(options) {
    // ...snip...
    this[MODULE_ID] = options.moduleId;
    this[MODULE_NAME] = options.moduleName;
  }

  // ...snip...

  get _moduleId() {
    // TODO: deprecate accessing _moduleId
    return this[MODULE_ID];
  }

  get _moduleName() {
    // TODO: deprecate accessing _moduleName
    return this[MODULE_NAME];
  }
}
```

Reply:
NOTE: I think the private symbol thing could be done in a separate PR (it does seem unrelated to this workingDir work)